src/Lights/Shadows/shadowGenerator.ts

```diff
@@ -888,6 +1154,88 @@ export class ShadowGenerator implements IShadowGenerator {
         });
     }
 
+    private _initializeCascadedShadowMaps(): void {
+        // Render target
+        let engine = this._scene.getEngine();
+        this._cascadeShadowMaps = [];
+        this._CSMtransformMatrices = [];
+        // for each cascade, create and set a light and shadowmap to said cascade
+        for (let index = 0; index < this.cascades; index++) {
+            if (engine.webGLVersion > 1) {
+                this._cascadeShadowMaps[index] = new RenderTargetTexture(this._light.name + "_cascadeShadowMap" + index, this._mapSize, this._scene, false, true, this._textureType, this._light.needCube(), undefined, false, false);
+                this._cascadeShadowMaps[index].createDepthStencilTexture(Constants.LESS, true);
+            } else {
+                this._cascadeShadowMaps[index] = new RenderTargetTexture(this._light.name + "_cascadeShadowMap" + index, this._mapSize, this._scene, false, true, this._textureType, this._light.needCube());
+            }
+            this._cascadeShadowMaps[index].wrapU = Texture.CLAMP_ADDRESSMODE;

```

Comment:
This creation should depend of the type of shadows e.g. simple/poisson/PCF and so on ?

Reply:
No idea, copy pasted from the ```_initializeShadowMap``` function, lol.

Reply:
Basically CSM is node a kind of shadow but is complementarty to the chosen shadow types.

You could for instance use PCF and cascades or Poisson sampling and CSM.

The initialization and setup should depend on it as well. If the function is a full copy besides the loop we should find a way to factorize it.

Reply:
ok, not entirely sure if this is what you wanted. But I factored out the six lines where the texture properties were inited with ```this._initializeRTTProperties(this._cascadeShadowMaps[index]);```, ditto for the default case.